<?xml version="1.0"?>
<launch>
  <arg name="debug" default="false"/>
  <arg name="required" default="true"/>
  <arg name="config" default="parameters_test.yaml"/>
  <arg name="problem_number" default="1"/>
  <arg name="problem" default="problem_test_$(arg problem_number)_a.yaml"/>
  <arg name="map" default="test_scenario_1.yaml"/>
  <arg name="planner_type" default="rrt"/>
  <arg name="n_tests" default="1"/>

  <!--Start the map server for the path planner-->
  <node name="map_server" pkg="map_server" type="map_server" args="$(find siar_planner)/test/$(arg map)" output="screen"/>

  <!-- <node name="process_costmap" pkg="siar_costmap" type="process_node" output="screen">
    <remap from="/costmap" to="/map"/>
    <param name="exp_decay" value="40"/>
    <param name="negative_obstacle" value="-128"/>
    <param name="positive_obstacle" value="127"/>
  </node> -->
  
  <node name="planner_test" pkg="siar_planner" type="planner_test" output="screen" if="$(arg debug)" launch-prefix="xterm -e gdb --args" >
    <!-- <remap from="/altitude_map" to="/processed_costmap"/> -->
    <remap from="/altitude_map" to="/map"/>
    <rosparam file="$(find siar_planner)/test/$(arg config)" command="load"  /> 
    <rosparam file="$(find siar_planner)/test/$(arg problem)" command="load"  /> 
    <param name="output_file" type="string" value="$(find siar_planner)/output_files/test_$(arg problem_number)_a_$(arg planner_type)_results.txt" />
    <param name="planner_type" value="$(arg planner_type)"/>
    <param name="n_tests" value="$(arg n_tests)"/>
  </node>
  <node name="planner_test" pkg="siar_planner" type="planner_test" output="screen" unless="$(arg debug)" required="$(arg required)" >
    <!-- <remap from="/altitude_map" to="/processed_costmap"/> -->
    <remap from="/altitude_map" to="/map"/>
    <rosparam file="$(find siar_planner)/test/$(arg config)" command="load"  /> 
    <rosparam file="$(find siar_planner)/test/$(arg problem)" command="load"  /> 
    <param name="output_file" type="string" value="$(find siar_planner)/output_files/test_$(arg problem_number)_a_$(arg planner_type)_results.txt" />
    <param name="planner_type" value="$(arg planner_type)"/>
    <param name="n_tests" value="$(arg n_tests)"/>
  </node>
  <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find siar_planner)/test/test.rviz"/> -->
   <!--<node pkg="rviz" type="rviz" name="rviz"/>-->
</launch>
